python_modules/dagster/dagster/_core/execution/plan/objects.py

```diff
@@ -110,10 +110,12 @@ def error_display_string(self) -> str:
         if self.error_source == ErrorSource.USER_CODE_ERROR:
             user_code_error = self.error.cause
             check.invariant(
-                user_code_error,
+                self.error.cls_name == "DagsterRedactedUserCodeError" or user_code_error,
                 "User code error is missing cause. User code errors are expected to have a"
                 " causes, which are the errors thrown from user code.",
             )
+            if not user_code_error:
+                return self.error.message.strip() + ":\n\n" + self.error.to_string()
```

Comment:
nit: suspicious seeing the invariant get updated but not the message text - can this be reordered instead or something? just take a second pass